lib/hanami/routing/route.rb

```diff
@@ -50,7 +50,8 @@ def generate(resolver, options = {}, &endpoint)
       # @since 0.2.0
       # @api private
       def nested_router
-        dest.routes if dest.respond_to?(:routes)
+        dest.routes if dest.respond_to?(:routes) &&
+          dest.routes.is_a?(Hanami::Router)
```

Comment:
After digging through the code base a bit more I believe we can do the more duck type style behavior checking with a check for responding to both `routes` and `routes.inspector` because it seems the next call on the `inspector` is `to_s` which everything has thanks to `Object`.

It is important to note that in scenarios where the nested route object does respond to `inspector` but the `to_s` of that object is different than that of `Routing::RoutesInspector` it may potentially make the `hanami routes` output look strange or screwed up. However, it does leave things more open and would actually allow other types of routes that are inspectable to be able to operate within `hanami routes`.

I will push the duck-type styled version up shortly.